packages/contracts/src/interest/DynamicDualRateContext.sol

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { EnumerableMap } from "@openzeppelin/contracts/utils/structs/EnumerableMap.sol";
+
+import { CalcInterestMetadata } from "@credbull/interest/CalcInterestMetadata.sol";
+import { IDynamicDualRateContext } from "@credbull/interest/IDynamicDualRateContext.sol";
+
+/**
+ * @title A dual rate context (full & reduced) where the Reduced Rates are configurable and many.
+ * @author credbull
+ * @notice A reference implementation of [IDynamicDualRateContext].
+ */
+contract DynamicDualRateContext is CalcInterestMetadata, IDynamicDualRateContext {
+    using EnumerableMap for EnumerableMap.UintToUintMap;
+    using Math for uint256;
+
+    error DynamicDualRateContext_InvalidPeriodRange(uint256 from, uint256 to);
+    error DynamicDualRateContext_InvalidReducedRatePeriod(uint256 from);
+
+    event DynamicDualRateContext_ReducedRateAdded(uint256 period, uint256 rateScaled, uint256 scale);
+    event DynamicDualRateContext_ReducedRateRemoved(uint256 period, uint256 rateScaled, uint256 scale);
```

Comment:
hi @jplodge-pro - please check yarn lint

./src/interest/DynamicDualRateContext.sol
  22:5  warning  Event name must be in CamelCase                                    event-name-camelcase
  23:5  warning  Event name must be in CamelCase                                    event-name-camelcase

Reply:
Yes, I saw that but thought there were pre-existing cases. It IS there for `error`s so I thought the same of `event`s. Thanks, will fix